bundle/edu.gemini.lchquery.servlet/src/main/scala/edu/gemini/lchquery/servlet/LchQueryFunctor.scala

```diff
@@ -180,11 +182,14 @@ class LchQueryFunctor(queryType: LchQueryFunctor.QueryType,
                 }
 
                 // Set the obs log comments.
-                obs.obsLogComments.filter(_.nonEmpty).foreach { obsLog =>
-                  setObsLogNode(new ObsLog() {
-                    obsLog.toList.foreach { case (lab,rec) =>
-                        setDataset(lab.toString)
-                        setRecord(rec.comment)
+                // If there are none, no setObsLogNode should be called.
+                obs.obsLogComments.filter(_.nonEmpty).foreach { log =>
+                  setObsLogNode(new ObsLogNode() {
+                    log.toList.foreach { case (lb,rc) =>
+                      getObsLog.add(new ObsLogRecord() {
+                        setId(lb.toString)
+                        setRecord(rc.comment)
+                      })
```

Comment:
Is the idea that they are only interested in comments?  If there are no comments they don't want dataset information?